Source/Editor/Surface/VisjectSurfaceWindow.cs

```diff
@@ -775,7 +775,34 @@ private void EditAttributesParameter(int index, Control label)
 
         private void DeleteParameter(int index)
         {
+            bool displayWarning = Editor.Instance.Options.Options.Interface.WarnOnDeletingUsedVisjectParameter;
+
             var window = (IVisjectSurfaceWindow)Values[0];
+            SurfaceParameter param = window.VisjectSurface.Parameters[index];
+
+            int connectedParameterNodeCount = 0;
+
+            List<SurfaceNode> nodes = window.VisjectSurface.Nodes;
+
+            for (int i = 0; i < window.VisjectSurface.Nodes.Count; i++)
+            {
+                if (nodes[i] is IParametersDependantNode node)
+                {
+                    if (displayWarning && (Guid)nodes[i].Values[0] == param.ID && nodes[i].GetBoxes().Any(b => b.Connections.Count > 0))
```

Comment:
I think it would be better to add method like `bool IsParamReferenced(SurfaceParameter param)` to `IParametersDependantNode ` so we could have more generic way of checking this parameter reference.

Reply:
Implemented that here: https://github.com/FlaxEngine/FlaxEngine/pull/3010/commits/57b4e9f2955b21d93239d718bc6e8fcaaf811906